### ISSUE TITLE
[BUG] Self signed cert in chain after upgrading to npm7

### ISSUE THREAD

**Issue by jorenbroekema** [npm/cli#2127]
### Current Behavior:
Self signed certificate in chain after upgrading to npm 7 from npm 6 latest.

```text
npm WARN registry Unexpected warning for https://registry.npmjs.org/: Miscellaneous Warning SELF_SIGNED_CERT_IN_CHAIN: request to https://registry.npmjs.org/@babel%2fparser failed, reason: self signed certificate in certificate chain
npm WARN registry Using stale data from https://registry.npmjs.org/ due to a request error during revalidation.
npm ERR! code SELF_SIGNED_CERT_IN_CHAIN
npm ERR! errno SELF_SIGNED_CERT_IN_CHAIN
npm ERR! request to https://registry.npmjs.org/@changesets%2fcli failed, reason: self signed certificate in certificate chain

npm ERR! A complete log of this run can be found in:
```

npm config list
```text
; "user" config from /Users/joren/.npmrc

@foo:registry = "https://foo.net/api/npm/" 
//foo.net/api/npm/:_authToken = (protected) 
cafile = "/Users/joren/certs/foo.ca-bundle" 
https_proxy = "http://localhost:3128" 
proxy = "http://localhost:3128/" 
registry = "https://foo.net/api/npm/" 
strict-ssl = true 

; "cli" config from command line options

omit = [] 
user-agent = "npm/7.0.8 node/v14.13.0 darwin x64" 

; node bin location = /usr/local/Cellar/node/14.13.0/bin/node
; cwd = /Users/joren/code/lion
; HOME = /Users/joren
; Run `npm config ls -l` to show all defaults.
```

Turning strict-ssl to `false` as a workaround also doesn't work, I get the same error. And I want strict-ssl to be on, of course :). 

If I downgrade back to npm 6, it works again.

### Expected Behavior:

npm 7 to handle my `~/.npmrc` cafile config properly. 

### Steps To Reproduce:
1. Use a proxy
2. Use any private NPM registry
3. Use my config as noted above, but edit the values to sync with your proxy and private NPM registry
4. Create some certificates and bundle them as one, add the path to this bundle to your npm config cafile
5. Verify things are working fine with npm 6
6. Upgrade to npm 7 (`npm i -g npm@7`)
7. npm install anywhere, see the error.

### Environment:

- OS: MacOS Catalina 10.15.5
- Node: 14.13.0
- npm: 7.0.8

**Comment by darcyclarke** [comment]
@jorenbroekema can you try updating to the latest v7 release (ie. `7.4.1`) & try replicating?

**Comment by jorenbroekema** [comment]
Sure, I'll update the issue next Monday after trying it with latest npm 7 👍

**Comment by jorenbroekema** [comment]
Updated to npm `7.4.2`, still getting the self signed certificate warning:

```
npm WARN registry Unexpected warning for https://registry.npmjs.org/: Miscellaneous Warning SELF_SIGNED_CERT_IN_CHAIN: request to https://registry.npmjs.org/@babel%2fcore failed, reason: self signed certificate in certificate chain
npm WARN registry Using stale data from https://registry.npmjs.org/ due to a request error during revalidation.
```
and then it fails because it cannot find a matching version for some dependency.

If I just do `npm info @babel/core` it kinda works, I still get the warning, but it manages to find it using stale data.

If I do `npm info` with some package for which stale data isn't there and it has to revalidate, I get:

```
npm ERR! code SELF_SIGNED_CERT_IN_CHAIN
npm ERR! errno SELF_SIGNED_CERT_IN_CHAIN
npm ERR! request to https://registry.npmjs.org/@web%2ftest-runner failed, reason: self signed certificate in certificate chain
```

Here's the complete log:

```
0 verbose cli [
0 verbose cli   '/usr/local/Cellar/node/14.13.0/bin/node',
0 verbose cli   '/usr/local/opt/node/bin/npm',
0 verbose cli   'info',
0 verbose cli   '@web/test-runner'
0 verbose cli ]
1 info using npm@7.4.2
2 info using node@v14.13.0
3 timing config:load:defaults Completed in 1ms
4 timing config:load:file:/usr/local/Cellar/node/14.13.0/lib/node_modules/npm/npmrc Completed in 0ms
5 timing config:load:builtin Completed in 0ms
6 timing config:load:cli Completed in 1ms
7 timing config:load:env Completed in 1ms
8 timing config:load:file:/Users/joren/code/lion/.npmrc Completed in 0ms
9 timing config:load:project Completed in 0ms
10 timing config:load:file:/Users/joren/.npmrc Completed in 1ms
11 timing config:load:user Completed in 1ms
12 timing config:load:file:/usr/local/Cellar/node/14.13.0/etc/npmrc Completed in 0ms
13 timing config:load:global Completed in 0ms
14 timing config:load:cafile Completed in 1ms
15 timing config:load:validate Completed in 1ms
16 timing config:load:setUserAgent Completed in 1ms
17 timing config:load:setEnvs Completed in 0ms
18 timing config:load Completed in 7ms
19 verbose npm-session 2c97b1d99d82aaec
20 timing npm:load Completed in 14ms
21 timing command:view Completed in 150ms
22 verbose type system
23 verbose stack FetchError: request to https://registry.npmjs.org/@web%2ftest-runner failed, reason: self signed certificate in certificate chain
23 verbose stack     at ClientRequest.<anonymous> (/usr/local/Cellar/node/14.13.0/lib/node_modules/npm/node_modules/minipass-fetch/lib/index.js:97:14)
23 verbose stack     at ClientRequest.emit (events.js:314:20)
23 verbose stack     at TLSSocket.socketErrorListener (_http_client.js:469:9)
23 verbose stack     at TLSSocket.emit (events.js:314:20)
23 verbose stack     at emitErrorNT (internal/streams/destroy.js:106:8)
23 verbose stack     at emitErrorCloseNT (internal/streams/destroy.js:74:3)
23 verbose stack     at processTicksAndRejections (internal/process/task_queues.js:80:21)
24 verbose cwd /Users/joren/code/lion
25 verbose Darwin 19.6.0
26 verbose argv "/usr/local/Cellar/node/14.13.0/bin/node" "/usr/local/opt/node/bin/npm" "info" "@web/test-runner"
27 verbose node v14.13.0
28 verbose npm  v7.4.2
29 error code SELF_SIGNED_CERT_IN_CHAIN
30 error errno SELF_SIGNED_CERT_IN_CHAIN
31 error request to https://registry.npmjs.org/@web%2ftest-runner failed, reason: self signed certificate in certificate chain
32 verbose exit 1
```

And my output for `npm config list`:

```
; "user" config from /Users/joren/.npmrc

@foo:registry = "https://foo.net/api/npm/" 
//foo.net/api/npm/:_authToken = (protected) 
cafile = "/Users/joren/certs/foo.ca-bundle" 
https_proxy = "http://localhost:3128" 
proxy = "http://localhost:3128/" 
strict-ssl = true 

; "cli" config from command line options

omit = [] 
user-agent = "npm/7.4.2 node/v14.13.0 darwin x64" 

; node bin location = /usr/local/Cellar/node/14.13.0/bin/node
; cwd = /Users/joren/code/lion
; HOME = /Users/joren
; Run `npm config ls -l` to show all defaults.
```

As you can see I have strict-ssl to true, and I refer to my bundled company certs through `cafile` which until now has always worked, and if I revert to npm 6 everything works just fine again.

**Comment by jorenbroekema** [comment]
Any updates on this?

**Comment by jorenbroekema** [comment]
https://github.com/npm/cli/issues/1700 noticed this one, seems like a duplicate, will test if my issue is fixed since 7.5.0
